hc/accounts/models.py

```diff
@@ -55,18 +55,28 @@ def set_api_key(self):
 
     def send_report(self):
         # reset next report date first:
+        report_period = ''
         now = timezone.now()
-        self.next_report_date = now + timedelta(days=30)
-        self.save()
+        if self.reports_allowed == '1':
+            self.next_report_date = now + timedelta(days=1)
+            report_period = "Daily"
+        elif self.reports_allowed == '2':
+            self.next_report_date = now + timedelta(days=7)
+            report_period = "Weekly"
+        elif self.reports_allowed == '3':
+            self.next_report_date = now + timedelta(days=30)
+            report_period = "Monthly"
```

Comment:
I think we should handle the case where the user hasn't enabled reports here (when reports_allowed is either 0 or not set)